public/controller/controller_functions.php

```diff
@@ -120,7 +120,8 @@ function check_semester(){
         }
 //FIXME: 
 //        header("Location: index.php?page=displaymessage&semester=".$semester."&course=".$course);
-        header("Location: ERROR_bad_semester_error.html");
+        header("Location: ERROR_Xbad_semester_error.html");
```

Comment:
I don't see the file "ERROR_Xbad_semester_error.html" anywhere in the website's DocumentRoot.  I also don't see the old file "ERROR_bad_semester_error.html", either.  Either header call would trigger an "HTTP 404 not found" error in Apache.